src/ui/index.js

```diff
@@ -77,12 +87,74 @@ export default async (kbnServer, server, config) => {
       },
       vars: defaults(app.getInjectedVars() || {}, uiExports.defaultInjectedVars),
     };
-
+    if (kibanaTranslations.length <= 0) {
+      const locale = await getTranslationLocale(acceptLanguages);
+      kibanaTranslations = await i18nPlugin.getRegisteredLocaleTranslations(locale);
+    }
     return this.view(app.templateName, {
       app: app,
       loadingGif: loadingGif,
       kibanaPayload: payload,
       bundlePath: `${config.get('server.basePath')}/bundles`,
+      welcomeMessage: translate('CORE-WELCOME_MESSAGE'),
+      welcomeError: translate('CORE-WELCOME_ERROR'),
```

Comment:
I was thinking we would expose the translate function itself to the jade template, and use it directly in the jade templates wherever translations are needed. That way we only have to worry about parsing jade templates when doing the verification step.